client/packages/system/src/Patient/CreatePatientModal/PatientFormTab.tsx

```diff
@@ -1,20 +1,45 @@
-import React, { FC } from 'react';
+import React, { FC, useMemo } from 'react';
 import {
+  BaseDatePickerInput,
   BufferedTextInput,
   InputWithLabelRow,
+  Select,
   Stack,
+  useFormatDateTime,
   useTranslation,
 } from '@openmsupply-client/common';
-import { useCreatePatientStore } from '../hooks';
+import { Gender, useCreatePatientStore } from '../hooks';
 import { PatientPanel } from './PatientPanel';
 
 export const PatientFormTab: FC<PatientPanel> = ({ patient, value }) => {
   const { updatePatient } = useCreatePatientStore();
   const t = useTranslation('patients');
 
+  const genderOptions = useMemo(
+    () =>
+      Object.keys(Gender).map(key => ({
+        label: key,
+        value: key,
+      })),
+    []
+  );
+  const dateFormatter = useFormatDateTime().customDate;
+
   return (
     <PatientPanel value={value} patient={patient}>
       <Stack spacing={2}>
+        <InputWithLabelRow
+          label={t('label.id')}
+          Input={
+            <BufferedTextInput
+              size="small"
+              sx={{ width: 250 }}
+              value={patient?.code ?? ''}
+              onChange={event => updatePatient({ code: event.target.value })}
+              autoFocus
+            />
+          }
+        />
```

Comment:
A suggestion to reduce the duplication here..
Outside of the component add this:
```
interface PatientInputRowProps {
  autoFocus?: boolean;
  field: keyof CreateNewPatient;
  label: string;
  patient?: CreateNewPatient;
  updatePatient: (patient: Partial<CreateNewPatient>) => void;
}

const PatientInputRow: FC<PatientInputRowProps> = ({
  autoFocus,
  label,
  field,
  patient,
  updatePatient,
}) => (
  <InputWithLabelRow
    label={label}
    Input={
      <BufferedTextInput
        size="small"
        sx={{ width: 250 }}
        value={patient?.[field] ?? ''}
        onChange={event => updatePatient({ [field]: event.target.value })}
        autoFocus={autoFocus}
      />
    }
  />
);
```

and then
```
<PatientInputRow
          label={t('label.id')}
          field="id"
          patient={patient}
          updatePatient={updatePatient}
        />
        <PatientInputRow
          label={t('label.first-name')}
          field="firstName"
          patient={patient}
          updatePatient={updatePatient}
          autoFocus
        />
        <PatientInputRow
          label={t('label.last-name')}
          field="lastName"
          patient={patient}
          updatePatient={updatePatient}
        />
```

Reply:
It only makes it a little bit shorter and slightly more complex and then there are the BaseDatePickerInput and Select input rows which don't fit this pattern. Do you mind leaving it as is?

Reply:
no, that's fine to leave as is